src/App.jsx

```diff
@@ -0,0 +1,34 @@
+import React, {useState} from "react";
+import { Controlled as CodeMirror } from "react-codemirror2";
+import "codemirror/lib/codemirror.css";
+import "codemirror/mode/javascript/javascript";
+
+import "./App.css";
+
+function App() {
+  const [value, setValue] = useState("// You can write the code");
+  const options = {
+    viewportMargin: Infinity,
+    mode: "javascript",
+    lineNumbers: true,
+    theme: "default height500 readonly"
+  };
+
+  const ejecuteCode = () => eval(value)
+
```

Comment:
So I think this will execute code (that has been stored in App state) after the button is clicked, but we don't actually do anything with it. There needs to be something that attaches the code to (for example) an active browser tab).

What I did (probably not the only answer) was:

set the permissions to the active tab (and any site) in the manifest: 
```
"permissions": ["https://*/*", "http://*/*", "tabs"],
```

I also had a background javascript file that ran under those conditions. Your version wouldn't use script.js, but the script could be passed there instead of a file:
```
chrome.browserAction.onClicked.addListener(function (tab) {
  chrome.tabs.executeScript(tab.ib, { file: "./script.js" });
})
```

Reply:
It is what he failed to understand or make work. What's more, I don't understand.
How to retain the code that the user writes? To run it and how I connect it to the browser screen. What you tell me you add to the Manifiest.json no?

On the other hand, how do you save this code that the user writes in the backgound.js

Reply:
I've never done a create-react-app chrome extension, but I'll try and clarify a few things. You might have to ask Monarch for anything beyond this though, I'm still a beginner too.

[So this](https://gist.github.com/danharper/8364399) was the simple example I based mine off of. It uses a .js file as input, instead of a code editor/text box like yours does. 

You won't be able to copy it exactly because it's not a CRA app, but it shows a manifest.json file. In it, some permissions are defined. The one called 'tabs' let me access the tabs that are open in the browser.

There is also a '**background**' section in the manifest.json. It has an array of scripts, one of which is '**background.js**'. This will run in the background, and where we can set up our '**listener**' code, to wait for certain events to happen (clicks, page loads, etc)

Our next piece of code is '**background.js**'. In the example a listener is added to the browserAction (the icon to the right of the address bar). It will activate when it is clicked, and takes a 'tab' object (the current opened tab) as a parameter. It then injects a javascript file to the open tab using **chrome.tabs.executeScript**. 

Instead of these lines
```
chrome.tabs.executeScript(tab.ib, {
		file: 'inject.js'
	});
```
try this instead:
```
chrome.tabs.executeScript(tab.ib, alert("test"));
```
And by substituting **alert('test')** with the code you want to run, you should be able to inject javascript into your page.

Let me know how it goes, and good luck!